src/control/lib/netdetect/netdetect_test.go

```diff
@@ -550,3 +558,53 @@ func TestValidateProviderSm(t *testing.T) {
 		AssertEqual(t, err, nil, "Network device configuration is invalid - provider not supported")
 	}
 }
+
```

Comment:
I have some ideas for modifying this a bit.  Currently, the GetDeviceClassStub() makes the directory and file entry for the netdev passed in, and then reads it back.  Then always returns the same sampleDevClass.  I think you might want to look at making one function that creates "/tmp/Ethernet", "/tmp/Infiniband", ... and populates that with the correct values for each device class.  Then, make a GetDeviceClassStub() that performs the ReadFile() from /tmp/ that you made instead of having it read from "/sys/class/net/%s/type".  Then, you would define function pointers in the functions that use GetDeviceClass so that they either point to GetDeviceClass() or GetDeviceClassStub().

Reply:
Thanks for suggestion. As you described, I created `WriteSampleType` method to create “/tmp/eth0”, “/tmp/eth1”, and so on, and populate with correct value. In the test, I create a map of `“eth0”: 1`, `“ib0”: 32`, and so on, instead of always using the constant value.

I don’t quite understand your last sentence. Does it mean that now we can use a function pointer that can point to either `GetDeviceClass()` or `GetDeviceClassStub()` in the production code (`netdetect.go`) and do better testing because now the production code can read from the `/tmp/<netdev>/type` that are guaranteed to exist instead of `/sys/class/net/<netdev>/type`?

Reply:
> Does it mean that now we can use a function pointer that can point to either GetDeviceClass() or GetDeviceClassStub() in the production code (netdetect.go) and do better testing because now the production code can read from the /tmp/<netdev>/type that are guaranteed to exist instead of /sys/class/net/<netdev>/type?

That is a correct interpretation.   Since this will require approval by others as well, let's ask @mjmac for feedback on the proposed approach in this PR before you get too deep into it.